worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/BukkitPlayer.java

```diff
@@ -190,9 +191,13 @@ public void teleport(Location location, String successMessage, String failMessag
         PaperLib.teleportAsync(getPlayer(), BukkitAdapter.adapt(location))
                 .thenApply(success -> {
                     if (success) {
-                        print(successMessage);
+                        if (successMessage != null && !successMessage.isEmpty()) {
+                            MessagingUtil.sendStringToChat(this, successMessage);
+                        }
                     } else {
-                        printError(failMessage);
+                        if (failMessage != null && !failMessage.isEmpty()) {
+                            printError(failMessage);
+                        }
```

Comment:
I don't think this needs changed?

Reply:
True it's not needed. This is just to be consistent. `MessagingUtil.sendStringToChat` is used on success, because it's configurable by a flag, the `failMessage` however isn't.

Reply:
I see, `MessagingUtil` handles macro replacements, and since the message can be disabled you need the isEmpty check to avoid sending a blank line.

I'd just do this, the comment clarifies why it's there too, reducing the chance it's removed by accident in the future.
```
if (success) {
  // The success message can be cleared via flag
  if (successMessage.isEmpty()) {
    ...
  }
} else {
  printError(failMessage);
}
```

Reply:
Done.